### PR TITLE
Detect wrong NetworkID in Transaction views

### PR DESCRIPTION
**Pull request by ThibautBremand**
How to test:
1/ Add a NetworkID in `packages/api/templates/index.html`, e.g.:
```
  function handleMintNFT() {
    GemWalletApi.isInstalled()
      .then(({ result }) => {
        if (result.isInstalled) {
          const transaction = {
            URI: '4d696e746564207468726f7567682047656d57616c6c657421',
            flags: {
              tfOnlyXRP: true,
              tfTransferable: true
            },
            // fee: '199',
            transferFee: 3000, // 3%,
            NFTokenTaxon: 0,
            networkID: 21338,
            memos: [
              {
                memo: {
                  memoType: '4465736372697074696f6e',
                  memoData: '54657374206d656d6f'
                }
              }
            ]
          };

          GemWalletApi.mintNFT(transaction)
            .then((res) => {
              console.log('Received response: ', res);
            })
            .catch((e) => {
              console.error('Cannot proceed the transaction: ', e);
            });
        }
      })
      .catch((e) => {
        console.error('GemWallet is not connected: ', e);
      });
  }
```

21338 for Xahau Testnet

2/ Open GemWallet, set the chain to XRPL.

3/ Try the transaction: a modal will appear asking to switch network

### CODE REVIEW

File: packages/extension/src/hooks/useTransactionStatus/useTransactionStatus.tsx [hunk 1]
```diff
@@ -134,15 +133,6 @@ export const useTransactionStatus = ({
           />
         );
       }
-      Sentry.captureException('Transaction failed - errorFees: ' + errorFees);
```
Comment (FlorianBouron) [inline packages/extension/src/hooks/useTransactionStatus/useTransactionStatus.tsx:137]: Why did you remove this?

Comment (ThibautBremand) [reply packages/extension/src/hooks/useTransactionStatus/useTransactionStatus.tsx:137]: I have changed the way the fees calculation errors are handled (better) and display (smoother). Also it's not relevant to track this with Sentry, especially since the fees calculation hook can throw errors when the hook has race conditions (see the comments in the code)

File: packages/extension/src/components/organisms/TransactionDetails/WrongNetworkIDModal/WrongNetworkIDModal.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  open,
+  onClose,
+  expectedNetwork,
+  currentNetwork,
+  setIsLoading
+}) => {
+  const { switchNetwork } = useNetwork();
+  const [showConfirmation, setShowConfirmation] = useState(false);
+
+  const displayNetwork = useCallback((network: NetworkNode) => {
```
Comment (FlorianBouron) [inline packages/extension/src/components/organisms/TransactionDetails/WrongNetworkIDModal/WrongNetworkIDModal.tsx:34]: No need to useCallback here as this is not resource-consuming.

Comment (FlorianBouron) [reply packages/extension/src/components/organisms/TransactionDetails/WrongNetworkIDModal/WrongNetworkIDModal.tsx:34]: I have resolved it as this is minor
